dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemoryStore.cs

```diff
@@ -0,0 +1,571 @@
+                                { "dimensions", this._dimensions }
+                            }
+                        }
+                    }
+                }
+            }
+        };
+    }
+
+    private async Task<IAsyncCursor<BsonDocument>> VectorSearchAsync(
+        int limit,
+        ReadOnlyMemory<float> embedding,
+        string collectionName,
+        CancellationToken cancellationToken
+    )
+    {
+        if (limit <= 0)
+        {
+            limit = int.MaxValue;
+        }
+
+        BsonDocument[] pipeline = Array.Empty<BsonDocument>();
+        switch (this._kind)
+        {
+            case "vector-ivf":
+                pipeline = GetVectorIVFSearchPipeline(embedding, limit);
+                break;
+            case "vector-hnsw":
+                pipeline = GetVectorHNSWSearchPipeline(embedding, limit);
+                break;
+        }
+
+        using var cursor = await this.GetCollection(collectionName)
+            .AggregateAsync<BsonDocument>(pipeline, cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+        return cursor;
+    }
+
+    private BsonDocument[] GetVectorIVFSearchPipeline(ReadOnlyMemory<float> embedding, int limit)
+    {
+        string searchStage =
+            @"
+        {
+            ""$search"": {
+                ""cosmosSearch"": {
+                    ""vector"": ["
+            + string.Join(
+                ",",
+                embedding.ToArray().Select(f => f.ToString(CultureInfo.InvariantCulture))
+            )
+            + @"],
+                    ""path"": ""embedding"",
+                    ""k"": "
+            + limit
+            + @"
+                },
+                ""returnStoredSource"": true
+            }
+        }";
+
+        string projectStage =
+            @"
+        {
+            ""$project"": {
+                ""similarityScore"": { ""$meta"": ""searchScore"" },
+                ""document"": ""$$ROOT""
+            }
+        }";
+
+        BsonDocument searchBson = BsonDocument.Parse(searchStage);
+        BsonDocument projectBson = BsonDocument.Parse(projectStage);
+        return new BsonDocument[] { searchBson, projectBson };
+    }
+
+    private BsonDocument[] GetVectorHNSWSearchPipeline(ReadOnlyMemory<float> embedding, int limit)
+    {
+        string searchStage =
+            @"
+        {
+            ""$search"": {
+                ""cosmosSearch"": {
+                    ""vector"": ["
+            + string.Join(
+                ",",
+                embedding.ToArray().Select(f => f.ToString(CultureInfo.InvariantCulture))
+            )
+            + @"],
+                    ""path"": ""embedding"",
+                    ""k"": "
+            + limit
+            + @",
+                    ""efSearch"": "
+            + this._efSearch
+            + @"
+                }
+            }
+        }";
+
+        string projectStage =
+            @"
```

Comment:
Also consider RAW Literal syntax:
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/#raw-string-literals

Reply:
Are you saying to convert it something like this

```
string projectStage = "{\"$project\": {\"similarityScore\": { \"$meta\": \"searchScore\" },\"document\": \"$$ROOT\"}}";
```

Reply:
@westey-m @dmytrostruk  I am having some issues with string interpolation using the triple quoted raw string literals. Is this blocking the PR right now? Can we revisit this at a later point. I'll add a TODO for this now. We have to start working on samples for Build, and need this PR to be merged in.

Reply:
Hey @aayush3011, no problem, this is not the most important.  I was actually just suggesting it for the static strings anyway.

Reply:
@westey-m Thanks for the review. Can we move forward with merging this PR?

Reply:
@aayush3011, I reviewed the .net side, and I have no further comments and I have approved.
@markwallace-microsoft, did we need someone to do a more in depth review of the Python code?